### ISSUE TITLE
Upgrade to gtk-4

### ISSUE THREAD

**Issue by kyle-leonhard** [amrbashir/menu-rs#259]
Hi! I happened to notice this crate depends on an unmaintained gtk3 crate. Are there plans to upgrade to gtk4 instead? https://github.com/gtk-rs/gtk4-rs

I see a handful of advisory issues were closed:
https://github.com/tauri-apps/muda/issues/258

I happened to noticed due to another issue: https://github.com/advisories/GHSA-wrw7-89jp-8q8g

**Comment by amrbashir** [comment]
We currently need gtk3 for creating tray icons, but there is some effort going on to move away from gtk to StatusNotifierItem protocol for trays, once that is done we can start migrating all of our stack to gtk4. 

This is probably a lot of work, especially for this menu crate and probably will land in the next major for Tauri (i.e v3)

**Comment by kyle-leonhard** [comment]
Thanks @amrbashir !
